### DISCUSSION TITLE
Would it be possible to issue new version/build with go1.18beta1?

### DISCUSSION THREAD

**Discussion by vearutop** [golangci/golangci-lint#2420] [Q&A] [answered]
Current build (compiled with go1.17.x) unsurprisingly fails on code that uses generics, however if I build `golangci-lint` from source
```
gotip install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```
it seems to work ok (at least in my case of configuration).

In CI I'm using `golangci/golangci-lint-action` and it presumably does not allow building with custom version of Go.

It would be great to use `golangci/golangci-lint-action` for code with type parameters.

**Comment by vearutop** [comment]
Though, I checked more and I see that `typecheck` feels unhealthy in generic codebases (flagging false positive unused imports). :(

**Reply by butuzov** [reply]
Just git clone ... && go build ... atm.

**Comment by ldez** [comment, marked as answer]
Golangci-lint now supports go1.18 (since v1.45.0)
